### PR TITLE
feat(tab): ts new UI

### PR DESCRIPTION
**Pull request by hund030**

### CODE REVIEW

File: tab/ts/default/src/components/sample/AzureFunctions.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import { Button, Image, Loader } from "@fluentui/react-northstar";
+import { useData } from "./lib/useData";
+import * as axios from "axios";
+import {
+  TeamsUserCredential,
+  getResourceConfiguration,
+  ResourceType,
+} from "teamsdev-client"
+
+var functionName = process.env.REACT_APP_FUNC_NAME || "myFunc";
```
Comment (henzhang-ms) [inline tab/ts/default/src/components/sample/AzureFunctions.tsx:11]: default name switch to "getUserProfile"

File: tab/ts/default/src/components/App.tsx [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
-function App() {
-  // Check for the Microsoft Teams SDK object.
-  if (microsoftTeams) {
-    return (
+export default function App() {
+  const { theme, loading } = useTeamsFx();
+  return (
+    <Provider
+      theme={theme || teamsTheme}
+      styles={{ backgroundColor: "#eeeeee" }}
```
Comment (henzhang-ms) [inline tab/ts/default/src/components/App.tsx:24]: I saw a background-color already in css .body

File: tab/ts/default/src/components/Tab.tsx.tpl [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import { Welcome } from "./sample/Welcome";
+
+export default function Tab() {
+  return <div>
+    <Welcome showFunction={ {{showFunction}} } />
+  </div>;
+}
```
Comment (henzhang-ms) [inline tab/ts/default/src/components/Tab.tsx.tpl:8]: may be new line required

File: tab/ts/default/src/components/sample/lib/useTeamsFx.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        {
+          type: ResourceType.API,
+          name: "default",
+          properties: {
+            endpoint: functionEndpoint,
+          }
+        }
+      ]
+    });
+      initialized = true;
```
Comment (henzhang-ms) [inline tab/ts/default/src/components/sample/lib/useTeamsFx.ts:37]: bad indent

File: tab/ts/default/src/components/sample/lib/useTeamsFx.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+// TODO fix this when the SDK stops hiding global state!
+let initialized = false;
+
+export function useTeamsFx() {
+  const [result] = useTeams({});
+  const { error, loading } = useData(async () => {
+    if (!initialized) {
+      loadConfiguration({
+      authentication: {
```
Comment (henzhang-ms) [inline tab/ts/default/src/components/sample/lib/useTeamsFx.ts:22]: looks bad indent

File: tab/ts/default/src/components/App.tsx (hunk 2 above)
Comment (hund030) [reply tab/ts/default/src/components/App.tsx:24]: We need to add it here. Seems `Provider` by default has another background color, which will overwrite the body's background color.

File: tab/ts/default/src/components/sample/lib/useTeamsFx.ts (hunk 5 above)
Comment (hund030) [reply tab/ts/default/src/components/sample/lib/useTeamsFx.ts]: Updated.

File: tab/ts/default/src/components/Tab.tsx.tpl (hunk 3 above)
Comment (hund030) [reply tab/ts/default/src/components/Tab.tsx.tpl]: Updated.

**Review by henzhang-ms — APPROVED** [review]
fast approve to avoid block TS scaffold.

**Pull request merged**